### PR TITLE
COP-3232: Add tests for voyage report forms mandatory field validation

### PR DESCRIPTION
**Pull request by Bothees**
## Description
This Pull request has e2e tests for validating error messages for mandatory fields on voyage report forms.

### How to test ?
npm run cypress:test -- --spec cypress/integration/sGMR/voyage-report-form-validation.spec.js

## Developer Checklist

\* Required

- [x] Up-to-date with master branch? \*

- [x] Does it have tests?

- [x] Pull request URL linked to JIRA ticket? \*

- [x] All reviewer comments resolved/answered? \*

Ensure you delete the branch once the Pull Request is merged.

### CODE REVIEW

**Review by p3dr0migue1 — APPROVED** [review]
Good job @Bothees

File: cypress/integration/sGMR/voyage-report-form-validation.spec.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  before(() => {
+    cy.task('readFileMaybe', 'cypress/fixtures/users.json').then((data) => {
+      const user = JSON.parse(data);
+      cy.checkUserExists(user.email).then((userExist) => {
+        if (userExist === false) {
+          cy.registerUser();
+        }
+      });
+    });
```
Comment (rafenden) [inline cypress/integration/sGMR/voyage-report-form-validation.spec.js:20]: I think we should use the new syntax :)

```suggestion
    cy.registerUser();
```

File: cypress/integration/sGMR/voyage-report-form-validation.spec.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  it('Should verify Arraival details mandatory data', () => {
+    const errors = [
+      'You must enter an arrival date',
+      'You must enter an arrival time',
+      'You must enter an arrival point',
+    ];
+    cy.enterDepartureDetails(departureDateTime, departurePort);
+    cy.saveAndContinue();
+    cy.url().should('include', '/page-2');
+    cy.get('.govuk-button[type="button"]').click();
```
Comment (rafenden) [inline cypress/integration/sGMR/voyage-report-form-validation.spec.js:75]: I suggest using the `cy.contains` function as it's easier to read and doesn't depend on CSS classes, e.g.
```suggestion
    cy.contains('button', 'Continue').click();
```

File: cypress/integration/sGMR/voyage-report-form-validation.spec.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    cy.saveAndContinue();
+    cy.url().should('include', '/page-5');
+    cy.saveAndContinue();
+    cy.get('.govuk-error-message').each((error, index) => {
+      cy.wrap(error).should('contain.text', errors[index]).and('be.visible');
+    });
+  });
+
+  afterEach(() => {
+    cy.exec('sh cypress/scripts/delete-reports.sh');
```
Comment (rafenden) [inline cypress/integration/sGMR/voyage-report-form-validation.spec.js:158]: At some point we should probably think of a better approach.

File: cypress/integration/sGMR/voyage-report-form-validation.spec.js (hunk 1 above)
Comment (Bothees) [reply cypress/integration/sGMR/voyage-report-form-validation.spec.js:20]: Yes making changes ! thanks

File: cypress/integration/sGMR/voyage-report-form-validation.spec.js (hunk 3 above)
Comment (Bothees) [reply cypress/integration/sGMR/voyage-report-form-validation.spec.js:158]: Yes right, We need to delete the reports as FE has the limitation now only to get max of 10 reports all together. So If not deleted, tests would start failing. I think if there is not limitation, we may not need to delete the reports. 
Or we should be approaching this issue differently.

**Review by rafenden — APPROVED** [review]

**Pull request merged**